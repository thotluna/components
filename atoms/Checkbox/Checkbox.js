import React, { useState } from 'react'
import PropTypes from 'prop-types'
import styles from './Checkbox.module.css'
import { getClasses } from '../../helpers/styles'

const handelChecked = ({ checked, setChecked, onChecked }) => () => {
  setChecked(!checked)
  onChecked(!checked)
}

export const Checkbox = ({ id, ariaLabel, isChecked, onChecked }) => {
  const [checked, setChecked] = useState(isChecked)
  const getStyles = getClasses(styles)({})
  return (
    <input
      id={id}
      aria-label={ariaLabel}
      type="checkbox"
      className={getStyles('checkbox')}
      checked={checked}
      onChange={handelChecked({ checked, setChecked, onChecked })}
    />
  )
}

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  ariaLabel: PropTypes.string.isRequired,
  isChecked: PropTypes.bool,
  onChecked: PropTypes.func,
}

Checkbox.defaultProps = {
  // isChecked: false,
}

export default Checkbox
