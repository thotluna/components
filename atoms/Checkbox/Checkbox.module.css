.checkbox {
  position: relative;
  display: inline-block;
  width: 28px;
  height: 28px;
  border: 2px solid var(--color-primary);
  -webkit-appearance: none;
  appearance: none;
  background-color: var(--color-background);
  border-radius: var(--border-radius-full);
}

.checkbox:checked::after {
  position: absolute;
  top: 0;
  left: 5px;
  color: var(--color-primary);
  content: '\2714';
  font-size: 18px;
}
