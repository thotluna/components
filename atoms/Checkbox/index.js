export { default } from './Checkbox'
export { options } from './constants'
export { default as styles } from './Checkbox.module.css'
