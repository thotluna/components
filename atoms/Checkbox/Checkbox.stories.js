import React from 'react'
import Checkbox from './Checkbox'

const Template = (args) => <Checkbox {...args} />

export default {
  title: 'Atoms/Checkbox',
  component: Checkbox,
  args: {
    id: 'check',
    ariaLabel: 'check',
    isChecked: false,
    onChecked: (checked) => console.log(checked),
  },
  argTypes: {
    isChecked: { control: 'boolean' },
  },
}

export const Default = Template.bind({})

export const Checked = Template.bind({})
Checked.args = { isChecked: true }
