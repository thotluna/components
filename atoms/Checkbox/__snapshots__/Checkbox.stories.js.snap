// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Checkbox Checked 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <input
    aria-label="check"
    checked={true}
    className="checkbox"
    id="check"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`Storyshots Atoms/Checkbox Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <input
    aria-label="check"
    checked={false}
    className="checkbox"
    id="check"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
